packages/libs/react-ui/src/components/Grid/GridContainer.tsx

```diff
@@ -0,0 +1,36 @@
+import { gapVariants, GridContainerClass } from './Grid.css';
+
+import classNames from 'classnames';
+import React, { CSSProperties, FC, ReactNode } from 'react';
+
+export interface IGridContainerProps {
+  children?: ReactNode;
+  spacing?: keyof typeof gapVariants;
+  templateRows?: CSSProperties['gridTemplateRows'];
+  templateColumns?: CSSProperties['gridTemplateColumns'];
+  templateAreas?: CSSProperties['gridTemplateAreas'];
+}
+
+const GridContainer: FC<IGridContainerProps> = ({
+  children,
+  spacing = 'md',
+  templateAreas,
+  templateRows,
+  templateColumns,
+}) => {
+  const styles = {
+    gridTemplateAreas: templateAreas,
+    gridTemplateRows: templateRows,
+    gridTemplateColumns: templateColumns,
+  };
+
+  const classList = classNames(GridContainerClass, gapVariants[spacing]);
+
+  return (
+    <div className={classList} style={styles}>
```

Comment:
I feel like using inline styles is generally looked down upon and I'm also not sure if there would be implications with server components when using this. I'm not sure if there is a better solution though. Do you have thoughts on this @sstraatemans ?

Reply:
yeah I would say, not use the style prop and keep it strict

Reply:
But, I can't think of an alternative? We don't have the flexibility of dynamic styling with VE since the styles are compiled to static style sheets. I believe other solutions like https://github.com/wayfair/rainbow-sprinkles do use inline styles so maybe this is the best option for this

Reply:
ok..as long as there is no style prop it is fine